commit aeff426f078d
Author: Brian Hackett <bhackett1024@gmail.com>
Date:   Tue Feb 6 14:11:34 2018 -0500

    Bug 1416523 - Don't attach arguments stub if the IC isn't monitored. r=jandem a=abillings
    
    --HG--
    extra : rebase_source : d1797b87843df0aea266bd7d1e0c98e9809b1291
    extra : source : 5fc563433fcfc2a08400988c247ccb23e45129d4
---
 js/src/jit/CacheIR.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git js/src/jit/CacheIR.cpp js/src/jit/CacheIR.cpp
index 22c0111756d8..aeef8f84d233 100644
--- js/src/jit/CacheIR.cpp
+++ js/src/jit/CacheIR.cpp
@@ -286,7 +286,7 @@ GetPropIRGenerator::tryAttachIdempotentStub()
         return true;
 
     // Object lengths are supported only if int32 results are allowed.
-    if ((resultFlags_ & GetPropertyResultFlags::AllowInt32) && tryAttachObjectLength(obj, objId, id))
+    if (tryAttachObjectLength(obj, objId, id))
         return true;
 
     // Also support native data properties on DOMProxy prototypes.
@@ -1433,6 +1433,9 @@ GetPropIRGenerator::tryAttachObjectLength(HandleObject obj, ObjOperandId objId,
     if (!JSID_IS_ATOM(id, cx_->names().length))
         return false;
 
+    if (!(resultFlags_ & GetPropertyResultFlags::AllowInt32))
+        return false;
+
     if (obj->is<ArrayObject>()) {
         // Make sure int32 is added to the TypeSet before we attach a stub, so
         // the stub can return int32 values without monitoring the result.
@@ -1697,6 +1700,9 @@ GetPropIRGenerator::tryAttachArgumentsObjectArg(HandleObject obj, ObjOperandId o
     if (!obj->is<ArgumentsObject>() || obj->as<ArgumentsObject>().hasOverriddenElement())
         return false;
 
+    if (!(resultFlags_ & GetPropertyResultFlags::Monitored))
+        return false;
+
     if (obj->is<MappedArgumentsObject>()) {
         writer.guardClass(objId, GuardClassKind::MappedArguments);
     } else {
